src/schema/posts.ts

```diff
@@ -1755,56 +1756,7 @@ export const resolvers: IResolvers<any, Context> = {
       { id, vote }: { id: string; vote: UserVote },
       ctx: Context,
     ): Promise<GQLEmptyResponse> => {
-      try {
-        if (!Object.values(UserVote).includes(vote)) {
-          throw new ValidationError('Unsupported vote type');
-        }
-
-        const post = await ctx.con.getRepository(Post).findOneByOrFail({ id });
-        await ensureSourcePermissions(ctx, post.sourceId);
-        const userPostRepo = ctx.con.getRepository(UserPost);
-
-        switch (vote) {
-          case UserVote.Up:
-            await userPostRepo.save({
-              postId: id,
-              userId: ctx.userId,
-              vote: UserVote.Up,
-              hidden: false,
-            });
-
-            break;
-          case UserVote.Down:
-            await userPostRepo.save({
-              postId: id,
-              userId: ctx.userId,
-              vote: UserVote.Down,
-              hidden: true,
-            });
-
-            break;
-          case UserVote.None:
-            await userPostRepo.save({
-              postId: id,
-              userId: ctx.userId,
-              vote: UserVote.None,
-              hidden: false,
-            });
-
-            break;
-          default:
-            throw new ValidationError('Unsupported vote type');
-        }
-      } catch (err) {
-        // Foreign key violation
-        if (err?.code === TypeOrmError.FOREIGN_KEY) {
-          throw new NotFoundError('Post or user not found');
-        }
-
-        throw err;
-      }
-
-      return { _: true };
+      return votePost({ ctx, id, vote });
```

Comment:
Would the userVote one not replace this one?

Reply:
Yes, but when frontend is refactored. I'll add a TODO